src/math/boolean-expression.ts

```diff
@@ -47,19 +19,62 @@ export function useBooleanExpressions() {
     return getters;
   }
 
+  function transform(ast: MathJson): MathJson<boolean> {
+    if (Array.isArray(ast)) {
+      const [functionName, ...args] = ast;
+      if (mathJsonOperatorMap.has(functionName as any)) {
+        return [functionName, ...args.map((v) => transform(v))];
+      } else if (functionName === "Error") {
+        return [functionName, ...args.map((v) => "" + v)];
+      } else {
+        return ["Error", `Unknown function ${functionName}`];
+      }
+    } else {
+      if (typeof ast === "string") {
+        const isFalse = /^(false|falsch|0)$/i.test(ast);
+        const isTrue = /^(true|wahr|1)$/i.test(ast);
+        if (isFalse) {
+          return false;
+        } else if (isTrue) {
+          return true;
+        } else {
+          // It's a variable
+          return ast;
+        }
+      } else if (ast?.type === "number") {
+        if (ast.value === "0") {
+          return false;
+        } else if (ast.value === "1") {
+          return true;
+        }
+      }
+      return ["Error", `Expected true or false: ${JSON.stringify(ast)}`];
+    }
+  }
+
+  function customPrinter(value: MathJson<boolean>): string | null {
+    if (value === true) {
+      return "\\mathtt{1}";
+    } else if (value === false) {
+      return "\\mathtt{0}";
+    } else {
+      return null;
+    }
+  }
+
   const mathJsonOperatorMap = new Map<MathJsonLogicalOperator, (a: boolean, b: boolean) => boolean>([
-    ["not", (a) => !a],
-    ["implies", (a, b) => !a || b],
-    ["and", (a, b) => a && b],
-    ["or", (a, b) => a || b],
-    ["xor", (a, b) => (a ? !b : b)],
-    ["nand", (a, b) => !(a && b)],
-    ["nor", (a, b) => !(a || b)],
-    ["equals", (a, b) => a == b],
-    ["subset", (a, b) => !b || a],
+    ["Not", (a) => !a],
```

Comment:
General question, was there a reason for the changing of all operator names from all lowercase to first letter uppercase?

Reply:
Good question, I suppose this is worth documenting. The reason for doing so was that I mimicked this MathJson https://cortexjs.io/math-json/